### PR TITLE
GEOMESA-2326 Update converter date formats to match Java DateTimeFormatter

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-convert/geomesa-convert-common/src/test/scala/org/locationtech/geomesa/convert2/transforms/ExpressionTest.scala [hunk 1]
```diff
@@ -231,17 +231,17 @@ class ExpressionTest extends Specification {
...
+      val exp = Expression("isoDate($1)")
+      exp.eval(Array("", "2015-01-01")).asInstanceOf[Date] must be equalTo testDate
     }
     "parse basicDate" >> {
       val exp = Expression("basicDate($1)")
       exp.eval(Array("", "20150101")).asInstanceOf[Date] must be equalTo testDate
     }
-    "parse isodatetime" >> {
-      val exp = Expression("isodatetime($1)")
-      exp.eval(Array("", "20150101T000000.000Z")).asInstanceOf[Date] must be equalTo testDate
```
Comment (jnh5y) [inline geomesa-convert/geomesa-convert-common/src/test/scala/org/locationtech/geomesa/convert2/transforms/ExpressionTest.scala:244]: Is it worth adding unit tests that show that these strings ("20150101", "20150101T000000.000Z") work with basicDate and basicDateTime?

Comment (elahrvivaz) [reply geomesa-convert/geomesa-convert-common/src/test/scala/org/locationtech/geomesa/convert2/transforms/ExpressionTest.scala:244]: i think those are already there, like the one right above this on line 239-240

Comment (jnh5y) [reply geomesa-convert/geomesa-convert-common/src/test/scala/org/locationtech/geomesa/convert2/transforms/ExpressionTest.scala:244]: Cool.  Merging.

**Comment by jnh5y** [comment]
Merged as db90f97f9b5b4b0f1183c4c2694d83e4e99e9e1d

**Pull request closed without merging**